### PR TITLE
replace deprecated method in StringUtils isEmpty with hasLength

### PR DESCRIPTION
**Pull request by Parisa68**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Unit/integration tests
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change

### CODE REVIEW

File: src/main/java/no/uio/ifi/localega/doa/services/StreamingService.java [hunk 1]
```diff
@@ -97,7 +97,7 @@ private InputStream getPlaintextResponse(byte[] header, InputStream bodyInputStr
         ByteArrayInputStream headerInputStream = new ByteArrayInputStream(header);
         SequenceInputStream sequenceInputStream = new SequenceInputStream(headerInputStream, bodyInputStream);
         Crypt4GHInputStream crypt4GHInputStream;
-        if (!StringUtils.isEmpty(startCoordinate) && !StringUtils.isEmpty(endCoordinate)) {
+        if (StringUtils.hasLength(startCoordinate) && StringUtils.hasLength(endCoordinate)) {
```
Comment (joshbaskaran) [inline src/main/java/no/uio/ifi/localega/doa/services/StreamingService.java:100]: While this is fine, we should probably just use `str.isBlank()` or `str.isEmpty()` instead depending on the use case.

File: src/main/java/no/uio/ifi/localega/doa/services/StreamingService.java [hunk 2]
```diff
@@ -109,7 +109,7 @@ private InputStream getPlaintextResponse(byte[] header, InputStream bodyInputStr
     private InputStream getEncryptedResponse(byte[] header, InputStream bodyInputStream, PrivateKey privateKey, String startCoordinate, String endCoordinate, String publicKey) throws GeneralSecurityException, IOException {
         PublicKey recipientPublicKey = KeyUtils.getInstance().readPublicKey(publicKey);
         Header newHeader = Crypt4GHUtils.getInstance().setRecipient(header, privateKey, recipientPublicKey);
-        if (!StringUtils.isEmpty(startCoordinate) && !StringUtils.isEmpty(endCoordinate)) {
+        if (StringUtils.hasLength(startCoordinate) && StringUtils.hasLength(endCoordinate)) {
```
Comment (joshbaskaran) [inline src/main/java/no/uio/ifi/localega/doa/services/StreamingService.java:112]: Same here as well.

File: src/main/java/no/uio/ifi/localega/doa/services/StreamingService.java (hunk 1 above)
Comment (joshbaskaran) [inline src/main/java/no/uio/ifi/localega/doa/services/StreamingService.java:100]:
```suggestion
        if (!startCoordinate.isBlank() && !endCoordinate.isBlank()) {
```
This is probably better given that it's part of core java.
Alternatively, you could also use `isEmpty()` method.

File: src/main/java/no/uio/ifi/localega/doa/services/StreamingService.java (hunk 2 above)
Comment (joshbaskaran) [inline src/main/java/no/uio/ifi/localega/doa/services/StreamingService.java:112]:
```suggestion
        if (!startCoordinate.isBlank() && !endCoordinate.isBlank()) {
```
Same as above.

**Review by joshbaskaran — CHANGES_REQUESTED** [review]

File: src/main/java/no/uio/ifi/localega/doa/services/StreamingService.java (hunk 1 above)
Comment (Parisa68) [reply src/main/java/no/uio/ifi/localega/doa/services/StreamingService.java:100]: Thank you for your suggestion. hasLength() method does the same job as the deprecated isEmpty(). but isBlank and isEmpty can behave differently. For example when the string is null, the methods from StringUtils can handle it but the methods from String class throw nullPointerException. also in the case of  whitespace strings, isBlank() does the opposite of what StringUtils.isEmpty() does.

**Review by joshbaskaran — APPROVED** [review]

**Pull request merged**